obal/data/modules/srpm.py

```diff
@@ -54,6 +56,27 @@ def copy_sources(spec_file, package_dir, sources_dir):
             run_command(["git-annex", "lock", "--force"])
 
 
+def fetch_remote_sources(source_location, source_system, sources_dir):
+    """
+    Copy RPM sources from a remote source like Jenkins to rpmbuild environment
+    """
+    if source_location and source_system:
+        url = "%s/lastSuccessfulBuild/artifact/*zip*/archive.zip" % (source_location)
```

Comment:
This structure implies `source_system == 'jenkins'`, but you never check that.

A possible solution could be having a global dict with systems and their url structure and use it here:

```python
SOURCE_SYSTEM_URLS = {
    'jenkins': '{}/lastSuccessfulBuild/artifact/*zip*/archive.zip',
}
…
def fetch_remote_sources(source_location, source_system, sources_dir):
    …
    url = SOURCE_SYSTEM_URLS[source_system].format(source_location)
```

And then you'd catch a `KeyError` below and tell the user "unknown source system"

Reply:
Excellent. I've implemented this tactic.